confmap/provider/httpprovider/provider.go

```diff
@@ -16,38 +16,87 @@ package httpprovider // import "go.opentelemetry.io/collector/confmap/provider/h
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
 	"fmt"
 	"io"
 	"net/http"
+	"os"
+	"path/filepath"
 	"strings"
 
 	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/confmap/provider/internal"
 )
 
+type TransportType string
+
 const (
-	schemeName = "http"
+	PlainText TransportType = "http"
+	TLS       TransportType = "https"
 )
 
 type provider struct {
-	client http.Client
+	client    http.Client
+	transport TransportType
 }
 
-// New returns a new confmap.Provider that reads the configuration from a file.
-//
-// This Provider supports "http" scheme, and can be called with a "uri" that follows:
+// New returns a new confmap.Provider that reads the configuration from a file using the configured transport mechanism.
+// There are two types of transport supported: PlainText (HTTP) and TLS (HTTPS).
 //
-// One example for http-uri be like: http://localhost:3333/getConfig
+// This Provider supports "http" and "https" schemes based on the transport chosen, and can be called with a "uri" that
+// follows:
 //
-// Examples:
-// `http://localhost:3333/getConfig` - (unix, windows)
-func New() confmap.Provider {
-	return &provider{client: http.Client{}}
+// One example for http-uri: http://localhost:3333/getConfig
+// One example for https-uri: https://localhost:3333/getConfig
+func New(transport TransportType) confmap.Provider {
+	return &provider{client: http.Client{}, transport: transport}
+}
+
+func (fmp *provider) configureTransport() error {
+	if fmp.transport == PlainText {
+		return nil
+	}
+
+	pool, err := x509.SystemCertPool()
+
+	if err != nil {
+		return fmt.Errorf("unable to create a cert pool: %w", err)
+	}
+
+	caCertPath := os.Getenv("OTEL_HTTP_PROVIDER_TLS_ADDITIONAL_CA_CERT")
```

Comment:
Good points. I will do some research before answering.

Reply:
I took a look into the resolver code and it does not pass any parameter to the confmap providers, only the URI whose resource should be retrieved. 

I agree with your points and the first point has special impact on `OTEL_HTTP_PROVIDER_TLS_INSECURE_SKIP_VERIFY`, because you might want to skip name verification for only some of the domain names being fetched. 

For `OTEL_HTTP_PROVIDER_TLS_ADDITIONAL_CA_CERT` it is ok to operate as a singleton because that is a public certificate that will be added to a pool of certificate containing the certificates installed in the system. In fact, we the latter variable could be a comma separated list of CA certificates.

I don't have a solution or suggestion on how to remediate these shortcomings. I'm considering attending the SIG meeting to gather feedback.

Reply:
An alternative might be to use the URI fragment to provide configuration to the provider.  The fragment is not supposed to be sent with an HTTP request anyways, so it should be safe to use for client configuration.

```
https://example.com/myConfig#insecure=true&certPath=/path/to/cert
```

Reply:
FYI, I created this proposal https://github.com/open-telemetry/opentelemetry-collector/issues/6892